gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql

```diff
@@ -0,0 +1,127 @@
+/* gpcontrib/arenadata_toolkit/arenadata_toolkit--1.4--1.5.sql */
+
+/*
+ This is part of arenadata_toolkit API for ADB Bundle.
+ This function collects information in db_files_current and db_files_history tables.
+ */
+CREATE OR REPLACE FUNCTION arenadata_toolkit.adb_collect_table_stats()
+RETURNS VOID
+AS $$
+DECLARE
+	ops oid[] DEFAULT (SELECT array_agg(oid) FROM pg_opclass WHERE opcname = 'timestamp_ops');
+BEGIN
+	IF NOT EXISTS (
+		SELECT
+		FROM pg_partition pp
+		JOIN pg_class cl1 ON pp.parrelid = cl1.oid
+		JOIN pg_partition_rule pr ON pr.paroid = pp.oid
+		JOIN pg_class cl2 ON cl2.oid = pr.parchildrelid
+		JOIN pg_namespace n1 ON cl1.relnamespace = n1.oid
+		JOIN pg_namespace n2 ON cl2.relnamespace = n2.oid
+		WHERE
+			CASE
+				WHEN pp.parclass[0] = ANY (ops)
+				THEN now() BETWEEN
+					CAST(substring(pg_get_expr(pr.parrangestart, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
+						AS TIMESTAMP WITHOUT TIME ZONE)
+					AND
+					CAST(substring(pg_get_expr(pr.parrangeend, pr.parchildrelid) FROM '#"%#"::%' FOR '#')
+						AS TIMESTAMP WITHOUT TIME ZONE)
+				ELSE FALSE
+			END
+			AND
+			cl1.relname = 'db_files_history'
+			AND
+			n1.nspname = 'arenadata_toolkit'
+			AND
+			n2.nspname = 'arenadata_toolkit'

```

Comment:
Do we really need to check child partition schema?

Reply:
As the filter is now changed to `pp.parrelid='arenadata_toolkit.db_files_history'::regclass`, I think it is no more relevant.